hydroflow_lang/src/graph/hydroflow_graph.rs

```diff
@@ -715,25 +715,90 @@ impl HydroflowGraph {
 
                             op_prologue_code.push(write_prologue);
                             subgraph_op_iter_code.push(write_iterator);
+
                             if include_type_guards {
-                                let fn_ident = format_ident!("check_{}", ident, span = op_span);
-                                let pull_push_trait = if is_pull {
+                                let source_info = {
+                                    // TODO: This crashes when running tests from certain directories because of diagnostics flag being turned on when it should not be on.
+                                    // Not sure of the solution yet, but it is not too important because the file is usually obvious as there can only be one until module support is added.
+                                    // #[cfg(feature = "diagnostics")]
+                                    // let path = op_span.unwrap().source_file().path();
+                                    #[cfg(feature = "diagnostics")]
+                                    let location = "unknown"; // path.display();
+
+                                    #[cfg(not(feature = "diagnostics"))]
+                                    let location = "unknown";
+
+                                    let location = location.to_string().replace(|x: char| !x.is_alphanumeric(), "_");
+
+                                    format!(
+                                        "{}__{}_{}__{}_{}",
+                                        location,
+                                        op_span.start().line,
+                                        op_span.start().column,
+                                        op_span.end().line,
+                                        op_span.end().column
+                                    )
+                                };
+                                let fn_ident = format_ident!("check_{}_{}_{}", ident, op_name, source_info, span = op_span);
+                                let type_guard = if is_pull {
```

Comment:
Lets also make separate fn_pull_ident and fn_push_ident and distinguish the two

Reply:
I did this by renaming "TypeGuard" type to Pull and Push, since the type of the new iterator also shows up in the profile.

Reply:
Oh ok